### PR TITLE
Feature Arrays with Queueable objects

### PR DESCRIPTION
**Pull request by zenangst**

### CODE REVIEW

**Comment by RamonGilabert** [comment]
Beautiful. :guardsman:

File: Source/Extensions/Array+Queueable.swift [hunk 1]
```diff
@@ -0,0 +1,17 @@
+public protocol Queueable {
+  func process() -> Bool
+}
+
+public extension Array where Element : Queueable {
+
+  public mutating func processQueue(from: Int = 0, _ to: Int? = nil, process: ((object: Element) -> Void)? = nil) {
+    let to = to != nil ? to! : count
+    let currentQueue = self[from..<to]
+    for (index, queuable) in currentQueue.enumerate() {
```
Comment (RamonGilabert) [inline Source/Extensions/Array+Queueable.swift]: for (index, queuable) in currentQueue.enumerate() where queuable.process() {

}

File: Source/Extensions/Array+Queueable.swift [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+      if queuable.process() == true {
+        process?(object: queuable)
+        removeAtIndex(index - (currentQueue.count - self.count))
```
Comment (vadymmarkov) [inline Source/Extensions/Array+Queueable.swift]: Could we drop `self`?

Comment (zenangst) [inline Source/Extensions/Array+Queueable.swift]: Fixed!

File: Source/Extensions/Array+Queueable.swift (hunk 1 above)
Comment (zenangst) [inline Source/Extensions/Array+Queueable.swift]: Nice addition! :cookie:

**Comment by RamonGilabert** [comment]
Beautiful.

**Pull request merged**